ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -340,10 +340,10 @@ mkApps
   => NodeKernel m remotePeer localPeer blk -- ^ Needed for bracketing only
   -> Tracers m remotePeer blk e
   -> Codecs blk e m bCS bCS bBF bBF bTX
-  -> Maybe DiffTime -- Workaround for #1882, test cases that can't cope with timeouts
+  -> (m (Maybe DiffTime))
```

Comment:
```suggestion
  -> m (Maybe DiffTime)
```

Reply:
I get that, it can then access "state" so, for example, it could read the current chain protocol parameters to calculate the appropriate timeouts